articles/functional-programming-in-java/index.md

```diff
@@ -0,0 +1,225 @@
+
+> [java.awt.Point[x=20,y=20]]
+
+As you can see, we can change an element using a reference pointing to the same object. To ensure that the List is genuinely immutable, it must only contain objects that are also immutable.
+
+Note Java 9 added a less verbose way to create these immutable Collections, following the basic syntax:
+
+```java
+
+List<String> strings = List.of("Hello", "World");
+
+```
+
+There are many other methods of the Collection subclasses of this fashion, such as Maps and Sets' corresponding methods.
+
+### Lambda Expressions
+A lambda expression is an expression used to create a method in-line. They have object-like qualities and we use them as arguments (in fact, they are their own objects, but we will explain that later). As an example of the syntax of these expressions, say we had this simple method:
+
+```java
+
+public static boolean isEven(int num){
+
+return num % 2 == 0;
+
+}
+
+```
+
+We can express this same logic as a lambda expression as follows:
+
+```java
+
+(num) -> { return num % 2 == 0; }
+
+```
+
+We write the arguments in brackets and the method body in curly brackets. These two separated with a `->`.
+
+Since there is only one argument within the brackets, we can simplify the expression:
+
+```java
+
+num -> { return num % 2 == 0; }
+
+```
+
+Since the method body only contains a return statement, we can simplify it even further:
+
+```java
+
+num -> num % 2 == 0
+
+```
+
+At this point, you may be wondering why we omit the types of the parameter and returned value. With Java being a statically typed language, it may not be very intuitive to you why this is the case. Yet, lambda expressions are strongly and statically typed, even though we don't show types. We will explain this in detail later.
+
+### Lambda Expressions vs Anonymous Inner Classes
+Lambda expressions, being a way to create "methods" in-line, draws some comparison to anonymous inner classes. As you may recall, anonymous inner-classes provide a way to build objects on the fly for use as arguments. We often use anonymous inner classes to pass event listeners into swing components:
+
+```java
+
+JButton b1 = new JButton("Hello World");
+
+b1.addActionListener(new ActionListener() {
+
+    public void actionPerformed(ActionEvent e) {
+
+        System.out.println("This is pretty cool");
+
+    }
+
+});
+
+```
+
+With the anonymous inner-class, we create the class in-line as it lacks any re-usability. Lambda expressions work almost the same way; they are also used to create single-use objects in-line. These objects are essentially instances of wrapper classes for the methods we use. Behind the scenes, lambda expressions are implementations of what is called `functional interfaces`.
+
+#### Functional Interfaces
+Functional interfaces are essentially interfaces that contain one unimplemented method. This allows the compiler to map lambda expressions into them. The idea is that they represent a specific functionality and not some kind of entity.
```

Comment:
You have used the `essentially` adverb a lot. Avoid too many adverbs as they make the article seem exaggerated.